lib/workers/repository/process/lookup/index.ts

```diff
@@ -263,6 +266,16 @@ export async function lookupUpdates(
         return res;
       }
       res.currentVersion = currentVersion!;
+      const currentVersionRelease = allVersions.find((v) =>
+        versioning.equals(v.version, currentVersion),
+      );
+      res.currentVersionReleaseTimeStamp =
+        currentVersionRelease?.releaseTimestamp ?? undefined;
+      // Reapply package rules to check matches for matchCurrentAge
+      config = applyPackageRules({
+        ...config,
+        currentVersionReleaseTimeStamp: res.currentVersionReleaseTimeStamp,
+      });
```

Comment:
Only run this applyPackageRules if:
- currentVersionReleaseTimestamp is defined
- Any of the packageRules contain matchCurrentAge